tests/test_resolver.py

```diff
@@ -49,21 +93,77 @@ async def fake(*args: Any, **kwargs: Any) -> List[Any]:
     return fake
 
 
+def fake_ipv6_addrinfo(hosts: Collection[str]) -> Callable[..., Awaitable[Any]]:
+    async def fake(*args: Any, **kwargs: Any) -> List[Any]:
+        if not hosts:
+            raise socket.gaierror
+
+        return [
+            (
+                socket.AF_INET6,
+                None,
+                socket.SOCK_STREAM,
+                None,
+                (h, 0, 0, 3 if ip_address(h).is_link_local else 0),
+            )
+            for h in hosts
+        ]
+
+    return fake
+
+
+def fake_ipv6_nameinfo(host: str) -> Callable[..., Awaitable[Any]]:
+    async def fake(*args: Any, **kwargs: Any) -> List[Any]:
+        return host, 0
+
+    return fake
+
+
+@pytest.mark.skipif(not gethostbyname, reason="aiodns 1.1 required")
+async def test_async_resolver_positive_ipv4_lookup(loop: Any) -> None:
+    with patch("aiodns.DNSResolver") as mock:
+        mock().getaddrinfo.return_value = fake_aiodns_getaddrinfo_ipv4_result(
+            ["127.0.0.1"]
+        )
+        resolver = AsyncResolver()
+        real = await resolver.resolve("www.python.org")
+        ipaddress.ip_address(real[0]["host"])
+        mock().getaddrinfo.assert_called_with(
+            "www.python.org",
+            family=socket.AF_INET,
+            flags=socket.AI_ADDRCONFIG,
+            port=0,
+            type=socket.SOCK_STREAM,
+        )
+
+
 @pytest.mark.skipif(not gethostbyname, reason="aiodns 1.1 required")
```

Comment:
This needs to be getaddrinfo as well as the other ones, but has to wait for a real release to change